### PR TITLE
fix(datasets): Various fixes

### PR DESCRIPTION
**Pull request by astrojuanlu**
## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Went from 362 seconds to 193 seconds @lrcouto. I cherry-picked @merelcht 's commit to pin Dask too. And the docs built successfully:

https://beta.readthedocs.org/projects/kedro-datasets/builds/23742913/

Feel free to take it over from here.

**Comment by lrcouto** [comment]
Are these changes related to the unit-tests failing? I'm trying to reproduce the error locally here but my local environment is not too trustworthy.

**Comment by astrojuanlu** [comment]
Unsure what's the hell is going on, but I attempted several fixes. The one that needs careful review 🔍⚠️  is the Tensorflow dataset changes I made.

- Use `uv` on RTD
- Pin Dask, both in test and in the requirements proper, because of #609 (proper term fix is allow pandas 2 for testing)
- Use smaller Hugging Face model to help the test pass + Use PyTorch-based model instead of TensorFlow-based one
- ⚠️ Rework TensorFlow dataset to work with Keras 3 (which deprecated `save_format` and relies on file extensions only) I didn't attempt to fix the `.h5` format so I'm expecting some tests to properly fail but at least the doctests should pass

I'm leaving this here

See #600

**Comment by astrojuanlu** [comment]
Managed to unblock RTD and the doctests

File: kedro-datasets/kedro_datasets/tensorflow/tensorflow_model_dataset.py
Comment (merelcht) [inline kedro-datasets/kedro_datasets/tensorflow/tensorflow_model_dataset.py:139]: 😂

**Review by merelcht — COMMENTED** [review]
I'm happy with most of these changes, but the Tensorflow ones seem a bit more involved. Changing the format from `h5` to `keras` seems breaking, so if this goes in we should immediately follow up on getting a proper fix before any release.

**Comment by astrojuanlu** [comment]
Indeed 👍🏽 I think to be completely honest the change to `.keras` extension is unwarranted. The culprit seems to be that we assume `.h5` are files and otherwise they're directories, and that trips tensorflow up, don't know why. The removal of the default save arguments is required though.

File: kedro-datasets/tests/tensorflow/test_tensorflow_model_dataset.py
Comment (astrojuanlu) [inline kedro-datasets/tests/tensorflow/test_tensorflow_model_dataset.py:196]: FYI

**Comment by astrojuanlu** [comment]
Looks like I fixed all the tests, at least on Ubuntu.

@DimedS @noklam @merelcht I removed a test that was expecting an error that is not erroring anymore, and removed an expected exception from another test that is not happening. That in principle looks like good news to me but it requires another pair of eyes.

I leave this to you now.

File: kedro-datasets/.readthedocs.yaml
Comment (astrojuanlu) [inline kedro-datasets/.readthedocs.yaml:14]: `uv` is at a newer version now, but "if it works, don't touch it"...

File: kedro-datasets/kedro_datasets/tensorflow/tensorflow_model_dataset.py
Comment (astrojuanlu) [inline kedro-datasets/kedro_datasets/tensorflow/tensorflow_model_dataset.py:65]: This is required for Keras 3, it's also removed from the tests

File: kedro-datasets/pyproject.toml
Comment (astrojuanlu) [inline kedro-datasets/pyproject.toml:36]: Probably we should add a comment here referencing #609

File: kedro-datasets/pyproject.toml
Comment (astrojuanlu) [inline kedro-datasets/pyproject.toml:235]: Previously `transformers` (Hugging Face) was using TensorFlow (since it's installed as part of the tests), but PyTorch support is more robust

File: kedro-datasets/tests/tensorflow/test_tensorflow_model_dataset.py
Comment (astrojuanlu) [inline kedro-datasets/tests/tensorflow/test_tensorflow_model_dataset.py:416]: Now we're not saving to a directory anymore

**Review by DimedS — APPROVED** [review]
Many thanks, @astrojuanlu . It looks good to me at first glance. I'll take a closer look at it today. For now, I'm approving it to unblock the PRs.

**Review by lrcouto — APPROVED** [review]

**Review by SajidAlamQB — APPROVED** [review]
This looks great, thank you @astrojuanlu!

**Comment by astrojuanlu** [comment]
This is going in

**Pull request merged**
